cli/crates/server/src/bridge/search.rs

```diff
@@ -0,0 +1,401 @@
+        let searcher = self.inner.reader()?.searcher();
+        let query_parser = QueryParser::for_index(&self.inner, self.default_fields.clone());
+        let tantivy_query = query_parser.parse_query(query)?;
+
+        let top_docs: Vec<(Score, DocAddress)> = searcher.search(&tantivy_query, &TopDocs::with_limit(limit))?;
+
+        let mut matching_records = vec![];
+        for (_score, doc_address) in top_docs {
+            // Retrieve the actual content of documents given its `doc_address`.
+            let retrieved_doc = searcher.doc(doc_address)?;
+            matching_records.push(
+                String::from_utf8(
+                    retrieved_doc
+                        .get_first(self.id_field)
+                        .and_then(tantivy::schema::Value::as_bytes)
+                        .expect("Id is always present")
+                        .to_vec(),
+                )
+                .expect("Id is a valid string"),
+            );
+        }
+
+        trace!(
+            "Found {} top documents:\n{:?}",
+            matching_records.len(),
+            matching_records
+        );
+        Ok(matching_records)
+    }
+
+    pub async fn build(pool: &SqlitePool, entity_type: &str, search_schema: &SearchSchema) -> Result<Index, ApiError> {
+        trace!("Building index for {entity_type} and schema:\n{search_schema:?}");
+
+        let mut schema_builder = Schema::builder();
+        let id_field = schema_builder.add_bytes_field("id", STORED);
+        let (fields, default_fields) = {
+            let mut all = vec![];
+            let mut defaults = vec![];
+            let mut search = vec![];
+
+            for field in &search_schema.fields {
+                let base_type = BASE_TYPE_RE.replace_all(&field.r#type, "");
+                search.push(SearchField {
+                    r#type: base_type.to_string(),
+                    ..field.clone()
+                });
+                let field = match base_type.borrow() {
+                    scalars::URL | scalars::EMAIL | scalars::PHONE => {
+                        schema_builder.add_text_field(&field.name, STRING)
+                    }
+                    scalars::STRING => schema_builder.add_text_field(&field.name, TEXT),
+                    scalars::DATE | scalars::DATE_TIME | scalars::TIMESTAMP => {
+                        schema_builder.add_date_field(&field.name, INDEXED)
+                    }
+                    scalars::INT => schema_builder.add_i64_field(&field.name, INDEXED),
+                    scalars::FLOAT => schema_builder.add_f64_field(&field.name, INDEXED),
+                    scalars::BOOL => schema_builder.add_bool_field(&field.name, INDEXED),
+                    scalars::IP => schema_builder.add_ip_addr_field(&field.name, INDEXED),
+                    unknown_type => {
+                        error!("Unknown field type: '{unknown_type}'");
+                        return Err(ApiError::ServerError);
```

Comment:
Wouldn't we want to return a specific error so a user can see it?

Reply:
That's really an internal error, it meant we had an issue with our parser. With my next PR I'm replacing the strings with a proper enum, so this will go away